File: python/Human.py
class Human:
    # Inisialisasi variabel untuk NIK, nama, dan jenis kelamin
    _nik = ""
    _nama = ""
    _jenis_kelamin = ""

    # Konstruktor untuk menginisialisasi variabel NIK, nama, dan jenis kelamin
    def __init__(self, nik='', nama='', jenis_kelamin=''):
        self._nik = nik
        self._nama = nama
        self._jenis_kelamin = jenis_kelamin

    # Getter untuk mendapatkan nilai NIK
    def getNik(self):
        return self._nik

    # Getter untuk mendapatkan nilai nama
    def getNama(self):
        return self._nama

    # Getter untuk mendapatkan nilai jenis kelamin
    def getJenisKelamin(self):
        return self._jenis_kelamin

File: cpp/Human.cpp
// // import library
// #include <iostream>
// #include <string>

// // deklarasi std agar tidak ditulis ulang
// using namespace std;

// kelas dasar Human, untuk merepresentasikan manusia
class Human {
    // variabel-variabel yang hanya dapat diakses oleh kelas turunan
    protected:
        string nik;
        string nama;
        string jenisKelamin;

    public:

        // konstruktor dengan parameter
        Human(string nik = "", string nama = "", string jenisKelamin = "") {
            this->nik = nik;
            this->nama = nama;
            this->jenisKelamin = jenisKelamin;
        }

        string getNik(){return nik;}
        string getNama(){return nama;}
        string getJenisKelamin(){return jenisKelamin;}

        // destruktor kosong
        ~Human(){

        }
};